sources/api/netdog/src/cli/write_resolv_conf.rs

```diff
@@ -0,0 +1,49 @@
+use super::{error, Result};
+use crate::dns::DnsSettings;
+use crate::{LEASE_FOLDER, PRIMARY_INTERFACE};
+use argh::FromArgs;
+use snafu::ResultExt;
+use std::fs;
+use std::path::{Path, PathBuf};
+
+#[derive(FromArgs, PartialEq, Debug)]
+#[argh(subcommand, name = "write-resolv-conf")]
+/// Writes /etc/resolv.conf, using DNS API settings if they exist
+pub(crate) struct WriteResolvConfArgs {}
+
+pub(crate) fn run() -> Result<()> {
+    // Use DNS API settings if they exist, otherwise use the primary interface's lease
+    let dns_settings =
+        if let Some(settings) = DnsSettings::from_config().context(error::DnsFromConfSnafu)? {
+            settings
+        } else {
+            let lease_path = primary_interface_lease()?;
+            DnsSettings::from_lease_file(lease_path).context(error::DnsFromLeaseSnafu)?
+        };
+    dns_settings
+        .write_resolv_conf()
+        .context(error::ResolvConfWriteFailedSnafu)?;
+    Ok(())
+}
+
+/// Return the path to the primary interface's ipv4/ipv6 lease
+fn primary_interface_lease() -> Result<PathBuf> {
+    let primary_interface = fs::read_to_string(PRIMARY_INTERFACE)
+        .context(error::PrimaryInterfaceReadSnafu {
+            path: PRIMARY_INTERFACE,
+        })?
+        .trim()
+        .to_lowercase();
+    let ipv4 = Path::new(LEASE_FOLDER).join(format!("leaseinfo.{}.dhcp.ipv4", primary_interface));
+    let ipv6 = Path::new(LEASE_FOLDER).join(format!("leaseinfo.{}.dhcp.ipv6", primary_interface));
+
+    // If both ipv4 and ipv6 leases exist, use the ipv4 lease for DNS settings
+    let ipv4_exists = Path::exists(&ipv4);
+    let ipv6_exists = Path::exists(&ipv6);
+    match (ipv4_exists, ipv6_exists) {
+        (true, true) => Ok(ipv4),
+        (true, false) => Ok(ipv4),
+        (false, true) => Ok(ipv6),
+        (false, false) => error::MissingLeaseSnafu { path: LEASE_FOLDER }.fail(),
+    }
```

Comment:
We entirely ignore the DHCP6 lease coming from the "install" path - it seems like we'd want the behavior to be consistent, otherwise we'll lose the IPv6 version of the settings the next time the lease is renewed and `resolv.conf` is rewritten.